Test and document saving movement datasets to netCDF (#606)
neuroinformatics-unit/movement: docs/source/user_guide/input_output.md

(target-io)=
# Input/Output

## Overview

Our goal with `movement` is to enable pipelines that are input-agnostic,
meaning they are not tied to a specific motion tracking tool or data format.
Therefore, our input/output functions are designed to facilitate data flows
between various third-party formats and `movement`'s own native

> [data structure](target-poses-and-bboxes-dataset) based on [xarray](xarray:).

Reviewer: For some reason I find that the bullet points come out of nowhere, so something like this may help it flow better:
```suggestion
[data structure](target-poses-and-bboxes-dataset) based on [xarray](xarray:). It may be useful to think of movement supporting two types of data loading/saving:
```

- [Supported third-party formats](target-supported-formats). `movement` provides convenient functions for loading/saving data in formats written by popular motion tracking tools as well as established data specifications. You can think of these as "Import" and "Export/Save As" functions.
- [Native saving and loading with netCDF](target-netCDF). `movement` leverages xarray's built-in netCDF support to save and load datasets while preserving all variables and metadata. **This is the recommended way to save your analysis state**, allowing your `movement`-powered workflows to resume exactly where they left off.

You are also welcome to try `movement` by loading some [sample data](target-sample-data) included with the package.

(target-supported-formats)=
## Supported third-party formats

`movement` supports the analysis of trajectories of keypoints (_pose tracks_) and of bounding box centroids (_bounding box tracks_),
which are represented as [movement datasets](target-poses-and-bboxes-dataset)
and can be loaded from and saved to various third-party formats.

| Source Software | Abbreviation | Source Format | Dataset Type |

> |-----------------|--------------|-------------|--------------|

Reviewer: Would it be helpful to add an extra column to state whether movement supports loading, saving, or both for each of these formats?

Author: Great idea! I've added such a column, with a footnote about "saving" to NumPy arrays.

![image](https://github.com/user-attachments/assets/7b171937-882f-4a03-aac9-89daf2a08b03)

Reviewer: Excellent

netCDF is a binary file format for self-described datasets that originated in the geosciences,
and netCDF files on disk directly correspond to {class}`xarray.Dataset` objects.

Saving to netCDF is the recommended way to preserve the complete state of your analysis,
including all variables, coordinates, and attributes.

To save any xarray dataset `ds` to a netCDF file:

```python
ds.to_netcdf("/path/to/my_data.nc")
````

To load the dataset back:

```python
import xarray as xr

ds = xr.open_dataset("my_data.nc")
```

Similarly, an {class}`xarray.DataArray` object (e.g. the `position` variable
of a `movement` dataset) can be saved to disk using the
{meth}`to_netcdf()<xarray.DataArray.to_netcdf()>` method, and loaded from disk using the
{func}`xarray.open_dataarray()` function.
As netCDF files correspond to Dataset objects,
these functions internally convert the DataArray to a Dataset before saving,
and then convert back when loading.

:::{note}
xarray also supports compression and chunking options with netCDF, which can be useful for managing large datasets.
For more details, see the [xarray documentation on netCDF](xarray:user-guide/io.html).
:::

> Below is a realistic example of how you may integrate netCDF into you

Reviewer: For some reason I don't like the specification of the example being "realistic". Are all the other examples unrealistic?
```suggestion
Below is an example of how you may integrate netCDF into you
```
